python_modules/dagster/setup.py

```diff
@@ -108,6 +108,7 @@ def get_version() -> str:
                 "pytest-xdist==2.1.0",
                 "pytest==6.1.1",
                 "responses==0.10.*",
+                "scikit-learn<1.0.0",  # scikit-learn 1.0 requires python 3.7
```

Comment:
Including the scikit-learn dependency here would pull in scikit-learn for everyone who depends on Dagster. If you put it in the setup.py under docs_snippets, we'd avoid that problem (although I think it's already there).